src/html/util.rs

```diff
@@ -83,39 +86,35 @@ impl NamespacedGlobalSymbols {
 
 pub type GlobalSymbolHrefResolver = Rc<dyn Fn(&[String], &String) -> String>;
 
+pub type ImportHrefResolver = Rc<dyn Fn(&[String], &String) -> Option<String>>;
+
 #[derive(Clone)]
 pub struct RenderContext<'ctx> {
-  tt: Rc<TinyTemplate<'ctx>>,
-  all_symbols: NamespacedSymbols,
+  pub(crate) ctx: &'ctx GenerateCtx<'ctx>,
+  current_exports: NamespacedSymbols,
+  current_imports: HashMap<String, String>,
```

Comment:
ideally it would re-use `GenerateCtx::global_symbols`, however first #427 needs to be investigated. Potentially easy to do, but perhaps conflicting/simplified by #426